build.gradle

```diff
@@ -1,3 +1,103 @@
+import groovy.json.JsonSlurper
+
 group 'com.amazonaws.blox'
 version '0.1-SNAPSHOT'
 description "Blox: Open Source schedulers for Amazon ECS"
+
+buildscript {
+    repositories {
+        maven { url 'https://plugins.gradle.org/m2/' }
+    }
+
+    dependencies {
+        classpath 'com.diffplug.gradle.spotless:spotless:2.4.1'
+    }
+}
+
+def unformattedProjects = [
+        'frontend-infrastructure',
+        'frontend-service-client'
+]
+
+allprojects {
+    apply plugin: 'com.diffplug.gradle.spotless'
+}
+
+configure(subprojects.findAll { !unformattedProjects.contains(it.name) }) {
+    spotless {
+        java {
+            googleJavaFormat()
+            licenseHeaderFile rootProject.file('licenses/license-header.java')
+        }
+    }
+}
+
+ext {
+    // This can be overridden by specifying -PawsProfile=my-other-profile
+    // on the Gradle command line:
+    awsProfile = project.hasProperty("awsProfile") ? project.awsProfile : "default"
+    println("Profile: ${awsProfile}")
+
+    awsCli = "/usr/local/bin/aws"
+    awsRegion = "us-west-2"
```

Comment:
Perhaps. I'm not sure how yet, though. Maybe wait until we see how we may want to reconfigure this?

(Gradle's only configuration mechanism is pretty cumbersome, you have to include `-pKey=Value` on the command line, there doesn't seem to be an easy way to set persistent local configuration).